File: VonexDealer.Vue/ClientApp/models/customer.js
//import { createAddress } from './Address';
//import { createContact } from './Contact';
//import { createName } from './Name';

export class Customer {
    constructor({
        customerID = 0,
        custNo = null,
        ubAccountNo = 0,
        phoneNumber = null,
        dealerID = 0,
        salutation = null,
        firstName = null,
        surname = null,
        primaryContactName = null,
        position = null,
        contactNo = null,
        phone = null,
        fax = null,
        mobile = null,
        email = null,
        dateAdded = new Date(),
        dob = null,
        planNo = 0,
        billRunCycleNumber = 0,
        contract_start = null,
        contract_end = null,
        contract_Term = null,
        groupNo = null,
        entityType = 0,
        addressID = null,
        companyID = null,
      
} = {}) {
        this.customerID = customerID;
        this.custNo = custNo;
        this.ubAccountNo = ubAccountNo;
        this.phoneNumber = phoneNumber;
        this.dealerID = dealerID;
        this.salutation = salutation;
        this.firstName = firstName;
        this.surname = surname;
        this.primaryContactName = primaryContactName;
        this.position = position;
        this.contactNo = contactNo;
        this.phone = phone;
        this.fax = fax;
        this.mobile = mobile;
        this.email = email;
        this.dateAdded = dateAdded;
        this.dob = dob;
        this.planNo = planNo;
        this.billRunCycleNumber = billRunCycleNumber;
        this.contract_start = contract_start;
        this.contract_end = contract_end;
        this.contract_Term = contract_Term;
        this.groupNo = groupNo;
        this.entityType = entityType;
        this.addressID = addressID;
        this.companyID = companyID;
    }
}

export function createCustomer(data) {
    //const address = createAddress(data.address);
    //const contacts = data.contacts.map(x => createContact(x));
    //const name = createName(data.name);

    return Object.freeze(new Customer(data));
}

File: VonexDealer.API/Models/Order/Customer.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace VonexDealer.API.Models.Order
{
   
    public class Customer
    {
       
        public Int64 CustomerID { get; set; }
        public string CustNo { get; set; }
        public Int64 UBAccountNo { get; set; }
        public string PhoneNumber { get; set; }
        public Int64? DealerID { get; set; }
        public string FirstName { get; set; }
        public string Surname { get; set; }
        public string PrimaryContactName { get; set; }
        public string Position { get; set; }
        public string ContactNo { get; set; }
        public string Phone { get; set; }
        public string Fax { get; set; }
        public string Mobile { get; set; }
        public string Email { get; set; }
        public DateTime? DateAdded { get; set; }
        public DateTime DOB { get; set; }
        public Int64? PlanNo { get; set; }
        public Int64? BillRunCycleNumber { get; set; }
        public string Contract_start { get; set; }
        public string Contract_end { get; set; }
        public string Contract_Term { get; set; }
        public string GroupNo { get; set; }
        public string EntityType { get; set; }

        public Int64? AddressID { get; set; }
        public virtual List<Address> AddressDetails { get; set; }

        public Int64? CompanyID { get; set; }
        public virtual Company CompanyDetails { get; set; }
        public string Salutation { get; set; }



    }

    public enum EntityType
    {
        Proprietor,
        SoleTrader,
        Consumer,
        RegisteredBusinessEntityOther
    }

}
